### PR TITLE
Allow users to resume applications later

### PR DESCRIPTION
**Pull request by stevenpetryk**
* Users are now greeted on the home page if they have an in-progress application, and are presented with a link to continue it.
* Users are now, after a short duration, sent an email reminding them to finish applying. This is configurable via `ENV["REMIND_USER_TO_FINISH_APPLYING_AFTER"]` as an ISO8601 duration.

### CODE REVIEW

**Comment by stevenpetryk** [comment]
Dis ready for review.

**Comment by maxcell** [comment]
We are missing `Dietary Restrictions (dietary_restrictions)`. So maybe we should consider this a part of Travel(?) or consider rewriting the section to just be Accommodations? But maybe we should consider this basic info and potentially remove the `phone number` from `Basic Info`? I don't actually know what we will do with the phone number. Maybe it won't be necessary now thinking about it in hindsight.

image

**Comment by maxcell** [comment]
Also small UI :bug:, when you go to the next role, it only has the Back button.
image

Along with this, when you hit Next button on that form, it will return you back to the Basic Info, is it intentional?
image

**Comment by maxcell** [comment]
I guess, I should reconsider this and actually just open issues. LMK what you think and I can revise the review.

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -3,6 +3,12 @@ class ApplicationController < ActionController::Base
 
   def home
     @countdown = Countdown.new
-    @user = User.new
   end
+
+  private
+
+  def current_user
+    @current_user ||= User.find_by(id: session[:user_id]) || User.new
```
Comment (maxcell) [inline app/controllers/application_controller.rb]: This is gives me chills as far as precedence, but tested it out and should be fine.

**Pull request merged**